### PR TITLE
CTCP-702: Added Arrivals CYA page and transition toggle

### PR DESCRIPTION
**Pull request by Reece-Carruthers**
- added arrivals CYA (includes identification, authorisations and location of goods **but not incidents**)
- split answers helpers out into different classes
- added `Mode` to all controllers so we can bounce around the journey from the final CYA page

### CODE REVIEW

**Comment by jackheslop96** [comment]
rebuild

File: app/controllers/CheckArrivalsAnswersController.scala [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+  def onPageLoad(mrn: MovementReferenceNumber): Action[AnyContent] = actions.requireData(mrn) {
+    implicit request =>
+      val sections = viewModelProvider(request.userAnswers).sections
+      Ok(view(mrn, sections))
+  }
+
+  def onSubmit(mrn: MovementReferenceNumber): Action[AnyContent] = actions.requireData(mrn) {
+    _ =>
+      ???
```
Comment (ajchilts) [inline app/controllers/CheckArrivalsAnswersController.scala:45]: I guess we just don't know this yet? Maybe put a // TODO in?

File: app/controllers/identification/authorisation/CheckAuthorisationAnswersController.scala [hunk 2]
```diff
@@ -17,32 +17,36 @@
...
 
 class CheckAuthorisationAnswersController @Inject() (
   override val messagesApi: MessagesApi,
+  navigatorProvider: AuthorisationsNavigatorProvider,
   actions: Actions,
   val controllerComponents: MessagesControllerComponents,
   view: CheckAuthorisationAnswersView,
   viewModelProvider: CheckAuthorisationAnswersViewModelProvider
-) extends FrontendBaseController
+)(implicit config: FrontendAppConfig)
```
Comment (ajchilts) [inline app/controllers/identification/authorisation/CheckAuthorisationAnswersController.scala:37]: Do we still need to be passing the FrontEndAppConfig in all these controllers etc?

**Review by ajchilts — APPROVED** [review]
Big PR! :-) 

On initial glance it looks OK. Few comments.

**Review by ajchilts — APPROVED** [review]
Looks good!

**Pull request merged**